File: exam/train/B-registration_plate.cpp
#include <iostream>
#include <string>
#include <vector>

bool conform(const std::string& evidence, const std::string& plate) {
    for (char c : evidence)
        if (plate.find(c) == std::string::npos)
            return false;
    return true;
}

int main() {
    int N;
    std::cin >> N;
    std::vector<std::string> evidences(N);
    for (auto& evidence : evidences)
        std::cin >> evidence;

    int M;
    std::cin >> M;
    int max_rank = 0;
    std::vector<std::string> plates(M);
    std::vector<int> ranks(M);
    for (size_t i = 0; i != M; ++i) {
        std::cin >> plates[i];
        for (const auto& evidence : evidences)
            if (conform(evidence, plates[i]))
                ++ranks[i];
        if (ranks[i] > max_rank)
            max_rank = ranks[i];
    }

    for (size_t i = 0; i != M; ++i)
        if (ranks[i] == max_rank)
            std::cout << plates[i] << "\n";
}

File: exam/2015/train/B-registration_plate.py
N = int(input())
data = []
for i in range(N):
    evidence = set(input())
    data.append(evidence)

M = int(input())
max_rank = 0
plates = []
for i in range(M):
    plate = input()
    plate_set = set(plate)
    rank = 0
    for evidence in data:
        if evidence.issubset(plate_set):
            rank += 1
    if rank > max_rank:
        max_rank = rank
    plates.append((plate, rank))

for plate, rank in plates:
    if rank == max_rank:
        print(plate)

